pkg/plugin/manager_unix_test.go

```diff
@@ -14,36 +14,57 @@ import (
 	"testing"
 	"time"
 
+	"github.com/go-git/go-git/v5"
 	"github.com/google/go-containerregistry/pkg/v1"
-	"github.com/sosedoff/gitkit" // Not work on Windows
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"github.com/aquasecurity/trivy/internal/gittest"
 	"github.com/aquasecurity/trivy/pkg/clock"
 	ftypes "github.com/aquasecurity/trivy/pkg/fanal/types"
 	"github.com/aquasecurity/trivy/pkg/log"
 	"github.com/aquasecurity/trivy/pkg/plugin"
 	"github.com/aquasecurity/trivy/pkg/utils/fsutils"
 )
 
-func setupGitServer() (*httptest.Server, error) {
-	service := gitkit.New(gitkit.Config{
-		Dir:        "./testdata",
-		AutoCreate: false,
-	})
+func setupGitRepository(t *testing.T, repo, dir string) *httptest.Server {
+	gs := gittest.NewServer(t, repo, dir)
 
-	if err := service.Setup(); err != nil {
-		return nil, err
-	}
+	worktree := t.TempDir()
+	r := gittest.Clone(t, gs, repo, worktree)
+
+	// git tag
+	gittest.SetTag(t, r, "v0.2.0")
+
+	// git commit
+	modifyManifest(t, worktree, "0.3.0")
+	gittest.CommitAll(t, r, "bump up to 0.3.0")
+
+	t.Log("git push")
+	err := r.Push(&git.PushOptions{})
+	require.NoError(t, err)
+
+	// git tag
+	gittest.SetTag(t, r, "v0.3.0")
 
-	ts := httptest.NewServer(service)
+	// git push --tags
+	gittest.PushTags(t, r)
 
-	return ts, nil
+	return gs
 }
 
-func TestManager_Install(t *testing.T) {
-	gs, err := setupGitServer()
+func modifyManifest(t *testing.T, worktree, version string) {
+	manifestPath := filepath.Join(worktree, "plugin.yaml")
+	b, err := os.ReadFile(manifestPath)
+	require.NoError(t, err)
+
+	b = bytes.ReplaceAll(b, []byte("0.2.0"), []byte(version))
+	err = os.WriteFile(manifestPath, b, 0644)
 	require.NoError(t, err)
+}
+
+func TestManager_Install(t *testing.T) {
+	gs := setupGitRepository(t, "test_plugin", "testdata/test_plugin")
 	t.Cleanup(gs.Close)
 
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I detected one problem with this test server:
this zip file doesn't contain nested dir:
- test_plugin.zip/plugin.yaml

But GitHub archive contains `repo name dir`:
check https://github.com/aquasecurity/trivy-plugin-kubectl/archive/refs/heads/main.zip
- trivy-plugin-kubectl-main.zip/trivy-plugin-kubectl-main/plugin.yaml

I think we should handle this case, but perhaps in another PR.

Reply:
Good point. It's not introduced in this PR. I'll open another PR.